drivers/ieee802154/ieee802154_dw1000.c

```diff
@@ -940,6 +947,46 @@ static int dwt_configure(const struct device *dev,
 	return -ENOTSUP;
 }
 
+/* driver-allocated attribute memory - constant across all driver instances */
+static const struct {
+	const struct ieee802154_phy_channel_range phy_supported_channel_ranges[2];
+	const struct ieee802154_phy_supported_channels phy_supported_channels;
+} attributes = {
+	.phy_supported_channel_ranges = {
+		{ .from_channel = 1, .to_channel = 5 },
+		{ .from_channel = 7, .to_channel = 7 },
+	},
+	.phy_supported_channels = {
+		.ranges = attributes.phy_supported_channel_ranges,
+		.num_ranges = 2U,
+	},
+};
```

Comment:
Hi @jfischer-no , thanks for your review. :-)

Sure, I can change those names to shorter ones, although the naming topic is much more involved than it may seem.

Those names have been chosen carefully in line with #61227 - so if we shorten them, we still need to work by these rules:
* Inside the driver distinction between PHY vs. MAC is mandatory unless the context is clear. Attributes can be defined in both contexts. Therefore "phy_attr" would be wrong.
* Names need to refer to IEEE 802.15.4 standard primitives if applicable. This is the case for the "channel" and "channel page" terms.
* Names must designate non-ambivalent points in the standard's global configuration space even if we only implement it partially for now. This PR was only necessary because we didn't follow that rule in the first place! This means that we need to give additional context to the "channel" term.

Additionally: Using abbreviations is not a mandatory coding guideline in Zephyr AFAICS. I personally consider them a maintainability problem on most occasions when I encounter them in Zephyr. After all we optimize maintainability (i.e. TCO) and not variable name length - the rest is up to proper IDE settings.

All that said, I do agree, though, that you've a point. My naming is inconsistent - thanks for pointing me to that.

I therefore changed

```
static const struct {
	const struct ieee802154_phy_channel_range phy_supported_channel_ranges[2];
	const struct ieee802154_phy_supported_channels phy_supported_channels;
} attributes ...
```

to

```
static const struct {
	const struct ieee802154_phy_channel_range phy_channel_range[2];
	const struct ieee802154_phy_supported_channels phy_supported_channels;
} attributes ...
```

(here and in the default DEFINE_... template)

Is there anything more you think we can still shorten, given the above information?

Reply:
TL;DR; Members of `static const struct` can be just `r` and `c`.  What matters is the name of parent `static const struct`.  `attributes` is too common in the context of a driver and should better be prefixed. `attr` is a widely used abbreviation in the tree.

Reply:
@jfischer-no As you insist: Prefixed and renamed _attributes_ to _drv_attr_.  For the rest you please first read then comment. TLDR not accepted.

Reply:
`drv_attr` is still not a correct name as is probably misleading, how I understood your patch,  struct ieee802154_phy_channel_range and struct ieee802154_phy_supported_channels are to describe phy/radio-controller capabilities and not drivers limitations. So this should be phy_attr or phy_attribute, and there is no reason to mix mac attributes in the same `static const structure` or how it could be not distinct where everything related is prefix with `ieee802154_phy_.`

Reply:
You seem not to be aware of #61227 which gives the context and explanation why this is correct (driver = PHY + hard MAC). Dismissed as wrong from a maintainer's perspective.

Reply:
Okay. I try to read it a few times, but TBH I cannot find this outcome. Further, the statement `driver = PHY + hard MAC` is very problematic. There is no driver for the HardMAC device in drivers/ieee802154, no matter what feature a radio controller has. I was under the assumption that it was cleaned up and layering improved, but I see it staying as it is.

Reply:
> There is no driver for the HardMAC device

The driver with most hard MAC features and the group from which the most concern re hardware offloading is actually (rightfully) coming, is Nordic's. As you seem to be working for them, I propose you co-ordinate yourself internally with @rlubos on that topic. He might be able to explain the Nordic requirements better to you. He's also been in the loop of our discussions all the time and should be able to explain the architectural direction to you as well as I can.